### PR TITLE
feat: add support for Daily Active User feature

### PR DESCRIPTION
**Pull request by mcollovati**
Adds support for Daily Active User feature for DAU enabled production build artifacts.

### CODE REVIEW

**Comment by mcollovati** [comment]
This is a draft PR, just to share a potential solution for DAU integration in Hilla and start a discussion.
Currently, it won't compile because it requires vaadin/flow#19771

File: packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java [hunk 1]
```diff
@@ -196,13 +214,22 @@ public ResponseEntity<String> serveEndpoint(
...
-            // Put a VaadinRequest in the instances object so as the request is
-            // available in the endpoint method
-            VaadinServletService service = (VaadinServletService) VaadinService
-                    .getCurrent();
-            CurrentInstance.set(VaadinRequest.class,
-                    new VaadinServletRequest(request, service));
+            if (service != null) {
+                DAUUtils.TrackableOperation.INSTANCE.execute(() -> {
+                    service.requestStart(vaadinRequest, vaadinResponse);
+                    DAUUtils.applyEnforcement(vaadinRequest, unused -> true);
```
Comment (mcollovati) [inline packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java:230]: Once the Flow PR will be merged, this should be changed int this way
```java
    if (DAUUtils.isDauEnabled(service)) {
        DAUUtils.applyEnforcement(vaadinRequest, unused -> true)
    }
```

Comment (mshabarov) [reply packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java]: Done.

File: packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java [hunk 2]
```diff
@@ -196,13 +241,13 @@ public ResponseEntity<String> serveEndpoint(
...
-            // available in the endpoint method
-            VaadinServletService service = (VaadinServletService) VaadinService
-                    .getCurrent();
-            CurrentInstance.set(VaadinRequest.class,
-                    new VaadinServletRequest(request, service));
+            enforcementResult = DAUUtils.trackDAU(this.vaadinService, request,
+                    response);
+            if (enforcementResult.isEnforcementNeeded()) {
+                return buildEnforcementResponseEntity(enforcementResult);
+            }
```
Comment (mshabarov) [inline packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java:250]: Does it make sense to run this block only if `DAUUtils.isDauEnabled(service)` is true?

**Review by mshabarov — DISMISSED** [review]

File: packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java (hunk 2 above)
Comment (tltv) [reply packages/java/endpoint/src/main/java/com/vaadin/hilla/EndpointController.java:250]: `VaadinRequest` wouldn't be set to `CurrentInstance` then and cleared afterwards. `DAUUtils.trackDAU` does that now. It was set there originally, so I would assume something to break in Hilla without it.

**Review by platosha — APPROVED** [review]

**Pull request merged**